holoviews/core/ndmapping.py

```diff
@@ -778,10 +779,8 @@ def label(self):
         if self._label:
             return self._label
         else:
-            labels = {v.label for v in self.values()
-                      if not v._auxiliary_component}
-            if len(labels) == 1:
-                return list(labels)[0]
+            if len(self):
+                return next(itervalues(self.data)).label
```

Comment:
Bottom layer should usually not be an annotation but I can just throw a while loop in there that'll iterate until it finds a non-auxiliar component.